packages/react/src/internal/menuButton/MenuButton.tsx

```diff
@@ -40,24 +45,31 @@ export type MenuButtonProps = React.PropsWithChildren<{
    * Spacing between the toggle button and the menu
    */
   menuOffset?: number;
+  /**
+   * Render label outside button and open menu via hover.
+   */
+  hoverAndClick?: boolean;
```

Comment:
Rename suggestion: withHoverOpen or something else?
Not important though, but we have boolean props named like withBorder etc